source/extensions/filters/http/bandwidth_limit/bandwidth_limit.cc

```diff
@@ -172,9 +209,18 @@ void BandwidthLimiter::updateStatsOnDecodeFinish() {
 
 void BandwidthLimiter::updateStatsOnEncodeFinish() {
   if (response_latency_) {
+    const auto& config = getConfig();
+
+    auto response_duration = response_latency_.get()->elapsed().count();
```

Comment:
We are planning to make stats write only, except for stat sinks. This code will end up broken in this case. Can a timestamp be used in this case?

Reply:
To get from the stats will give me a consistent delay value for response trail and the stats. 
Make stats write only seems a challenge work, can we read the stats here before you have a concrete plan how to remove the stats read reference.